storagemarket/impl/provider.go

```diff
@@ -581,25 +584,28 @@ func (p *Provider) start(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("Migrating storage provider state machines: %w", err)
 	}
-	if err := p.restartDeals(); err != nil {
-		return fmt.Errorf("Failed to restart deals: %w", err)
-	}
-	return nil
-}
 
-func (p *Provider) restartDeals() error {
 	var deals []storagemarket.MinerDeal
-	err := p.deals.List(&deals)
+	err = p.deals.List(&deals)
 	if err != nil {
-		return err
+		return xerrors.Errorf("failed to fetch deals during startup: %w", err)
 	}
+	if err := p.shardReg.registerShards(ctx, deals); err != nil {
+		return fmt.Errorf("Failed to restart deals: %w", err)
+	}
```

Comment:
We currently form a single batch. Is the intent to form multiple batches? Why do we need this batching? The DAG store will return from `RegisterShard` quickly.

Reply:
Makes sense, my assumption was that we needed to wait until there is a signal on the channel returned from RegisterShard before we can call the migration complete.
As discussed sync I will remove the batching as we only need to wait until RegisterShard returns (not until there is a signal on the channel)